src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -517,7 +517,304 @@ public with sharing class ADDR_Addresses_TEST {
+                            ' undeliverable status should not match the address undeliverable status when the address' +
+                            ' is not marked as the default address.');
+            for (Contact contact : updatedAccount.Contacts) {
+                System.assert(contact.is_Address_Override__c == false, 'The contact should not be marked with an ' +
+                        'address override since that will cause inaccuracies in this test scenario.');
+                System.assertEquals(address.Undeliverable__c, contact.Undeliverable_Address__c, 'The contact ' +
+                        'undeliverable address status should match the default address undeliverable status when the ' +
+                        'contact address override value is false.');
+            }
+        }
+    }
+
+    @isTest
+    private static void updatedUndeliverableAddressShouldSyncToAccountAndHouseholdContactOverrides() {
+        createTestData(2, 2);
+        Address__c overrideAddressToCreate = new Address__c();
+            overrideAddressToCreate.Address_Type__c = 'Home';
+            overrideAddressToCreate.MailingStreet__c = 'Street Override';
+            overrideAddressToCreate.MailingCity__c = 'City Override';
+            overrideAddressToCreate.MailingState__c = 'Washington';
+            overrideAddressToCreate.MailingPostalCode__c = 'Zip Override';
+            overrideAddressToCreate.MailingCountry__c = 'United States';
+            overrideAddressToCreate.Household_Account__c = testAccounts[0].Id;
+        insert overrideAddressToCreate;
+
+        Contact contact = new Contact(
+            FirstName = 'Test Contact',
+            LastName = 'Override',
+            is_Address_Override__c = true,
+            AccountId = testAccounts[0].Id,
+            MailingState = 'Washington',
+            MailingCity = 'City Override',
+            MailingStreet = 'Street Override',
+            MailingPostalCode = 'Zip Override',
+            MailingCountry = 'United States',
+            npe01__Primary_Address_Type__c = 'Home'
+        );
+        insert contact;
+
+        Address__c overrideAddress = [SELECT Undeliverable__c
+                                      FROM Address__c
+                                      WHERE MailingPostalCode__c = 'Zip Override'];
+
+        Test.startTest();
+            overrideAddress.Undeliverable__c = true;
+            update overrideAddress;
+        Test.stopTest();
+
+        Contact updatedContact = [SELECT Undeliverable_Address__c, is_Address_Override__c
+                                  FROM Contact
+                                  WHERE Id = :contact.Id];
+        System.assert(updatedContact.is_Address_Override__c == true);
+        System.assertEquals(overrideAddress.Undeliverable__c, updatedContact.Undeliverable_Address__c);
+    }
+
+    @isTest
+    private static void updatedUndeliverableDefaultAddressShouldSyncToContactOverrideNonHousehold() {
```

Comment:
I'd recommend naming the method for what you are testing - updatedUndeliverableDefaultAddressShouldSyncToContactOverrideOrganizationAccount??